lib/bolt/inventory.rb

```diff
@@ -36,6 +36,12 @@ def initialize(target)
       end
     end
 
+    class UnknownProtocolError < Bolt::Error
+      def initialize(uri)
+        super("Unknown protocol found in #{uri}", 'bolt.inventory/unknown-protocol')
```

Comment:
Could this come from the config too? Should it be bolt.config maybe?(nitpick)

Reply:
Yeah it could, have a suggestion how I should clean it up?

Reply:
`bolt.config/unknown-transport` seems more general and slightly clearer to the user.

Reply:
Actually, this is currently only thrown if the target URI itself specifies an unknown protocol. I haven't dealt with unexpected input from config/inventory.

Reply:
ah so this could come from the --nodes or nodes listed in the inventory.
`bolt.target/unknown-transport` or just `bolt/unknown-transport` it's probably just fine as is.